perf-tests/clusterloader2/pkg/framework/config/client_config.go

```diff
@@ -35,16 +38,62 @@ const (
 	burst       = 200
 )
 
+var muxInitializedMultiClientConfig sync.Mutex
+
+var extraAPIServerAddresses []string
+
+func SetAPIServers(apiServerAddresses string) error {
+	muxInitializedMultiClientConfig.Lock()
+	defer muxInitializedMultiClientConfig.Unlock()
+	if len(extraAPIServerAddresses) > 0 {
+		return fmt.Errorf("Api Server addresses already set [%v]", extraAPIServerAddresses)
+	}
+
+	addrs := strings.Split(apiServerAddresses, ";")
+	for _, item := range addrs {
+		extraAPIServerAddresses = append(extraAPIServerAddresses, item)
+	}
+
+	if len(extraAPIServerAddresses) == 0 {
+		klog.Infof("Passed in api server addresses [%s]. Keep kubeconfig value instead.", apiServerAddresses)
+	}
+	return nil
+}
+
 // PrepareConfig creates and initializes client config.
 func PrepareConfig(path string) (*restclient.Config, error) {
+	muxInitializedMultiClientConfig.Lock()
+	defer muxInitializedMultiClientConfig.Unlock()
+
 	config, err := loadConfig(path)
 	if err != nil {
 		return nil, err
 	}
 	if err = initializeWithDefaults(config); err != nil {
 		return nil, fmt.Errorf("config initialization error: %v", err)
 	}
-	return config, nil
+
+	if len(extraAPIServerAddresses) == 0 {
+		return config, nil
+	}
+
+	// create config for additional api server partitioning
+	aggConfig := restclient.NewAggregatedConfig(config.GetConfig())
+	for _, apiServerAddr := range extraAPIServerAddresses {
+		newConfig, _ := loadConfig(path)
+		initializeWithDefaults(newConfig)
+		c := newConfig.GetConfig()
+
+		// format: https://34.1.1.1
+		c.Host = apiServerAddr
+		aggConfig.AddConfig(c)
+	}
+
+	for i, config := range aggConfig.GetAllConfigs() {
+		klog.Infof("Kubeconfig %d host %v", i, config.Host)
```

Comment:
The log message will be hard to understand. A reader probably would be puzzled by the following log message:

"kubeconfig 0 host https://34.1.1.1"

Reply:
This is used by perf test only. Not for end user.